### PR TITLE
Request improvements

### LINKED ISSUE #138: soft delete pto requests
deleted pto requests need to leave an easily auditable paper trail

### LINKED ISSUE #139: deleting requests triggers refund mechanism
when a pto request is deleted, all requests made after the deleted requests should be checked if a lower price _would have_ been paid if the deleted request was never made. If so, associated user account should be refunded the difference.

### LINKED ISSUE #155: add soft delete check to sync

### PR DESCRIPTION
**Pull request by merchang**
### Test Plan
**verify requests are soft deleted**
- [x] delete a from the side bar on the calendar page
- [x] verify the request no longer appears in the UI: 1. the sidebar 2. user#show 3. user#edit
- [x] verify the user was refunded appropriately
- [x]  verify the users information was removed successfully from the related calendar object (signed_up_total, signed_up_agents[])
- [x] locate the request in console, verify `is_deleted => true`

**verify recalculation**
- [x] create a *new* request for user A
  - this request should for a calendar day that is on the cusp of a price increase, i.e. this request will move the pushes the day to a higher per hour price. (update the base_value of an item and manually run UpdatePrice.update_calendar_item() in console if needed)
- [x] create a *new* request for user B on the same day
- [x] delete the request for user A in the UI
- [x] verify user b's request price went down appropriately
- [x] verify user b's bank value was refunded appropriately

**odds and ends**
- [x] verify deleted requests appear in request export csv (for all users and individuals)
- [x] verify requests soft delete and refund when deleting as an admin/sup

---

fixes #138 
fixes #139 
fixes #155

### CODE REVIEW

**Comment by StoveSolder** [comment]
https://cl.ly/02eb6f12c04d

Deleting request via sidebar still destroys it, cannot be found through console

https://cl.ly/42cde47da767

File: app/controllers/pto_requests_controller.rb [hunk 1]
```diff
@@ -126,12 +171,11 @@ def destroy
             if @pto_request.reason == 'make up / sick day'
                 return sub_make_up_day
             end 
-
             remove_request_info
             RequestsMailer.with(user: @user, pto_request: @pto_request).delete_request_email.deliver_now
             return redirect_to root_path, notice: "Your request was deleted"
```
Comment (StoveSolder) [inline app/controllers/pto_requests_controller.rb]: redirecting to the root_path is kinda jarring, not expecting to go back to the calendar after deleting a user's request.

File: app/controllers/pto_requests_controller.rb [hunk 2]
```diff
@@ -126,12 +171,11 @@ def destroy
...
         else
-            return redirect_to root_path, notice: "Somthing went wrong"
+            return redirect_to root_path, notice: "Something went wrong"
```
Comment (StoveSolder) [inline app/controllers/pto_requests_controller.rb]: Plz to make bad things 'alert:' instead of 'notice:' Good things can stay 'notice:'

**Comment by StoveSolder** [comment]
https://cl.ly/1605722a3e86 ✔️

File: app/controllers/pto_requests_controller.rb (hunk 1 above)
Comment (merchang) [reply app/controllers/pto_requests_controller.rb]: should be resolved by 05f8385

File: app/controllers/pto_requests_controller.rb (hunk 2 above)
Comment (merchang) [reply app/controllers/pto_requests_controller.rb]: should be resolved by 05f8385

**Pull request merged**